packages/suite/src/components/suite/modals/ReduxModal/CardanoWithdrawModal.tsx

```diff
@@ -0,0 +1,156 @@
+    display: flex;
+    align-items: center;
+    width: 100%;
+    margin-top: 24px;
+    justify-content: center;
+`;
+
+export const Text = styled.div`
+    color: ${({ theme }) => theme.legacy.TYPE_LIGHT_GREY};
+    margin-bottom: 8px;
+    margin-top: 8px;
+    ${typography.hint}
+`;
+
+export const Content = styled.div`
+    display: flex;
+    overflow: hidden;
+    padding-left: 10px;
+    margin-left: -10px;
+    width: 100%;
+`;
+
+export const Value = styled.div`
+    ${typography.hint}
+    color: ${({ theme }) => theme.legacy.TYPE_DARK_GREY};
+    font-variant-numeric: tabular-nums slashed-zero;
+    width: fit-content;
+    background: ${({ theme }) => theme.legacy.BG_LIGHT_GREY};
+    border: 1px solid ${({ theme }) => theme.legacy.STROKE_GREY};
+    border-radius: ${borders.radii.xs};
+    word-break: break-all;
+    padding: 10px;
+    display: flex;
+`;
+
+type Props = {
+    onCancel: () => void;
+};
+
+export const CardanoWithdrawModal = ({ onCancel }: Props) => {
+    const { voteAbstain, voteDelegate } = useCardanoStaking();
+    const account = useSelector(state => state.wallet.selectedAccount.account);
+    if (!account || account.networkType !== 'cardano') {
+        throw Error('CardanoWithdrawModal used for other network');
+    }
+
+    const cardanoNetwork = account.symbol === 'ada' ? 'mainnet' : 'preview';
+    const { trezorDRep } = useSelector(state => state.wallet.cardanoStaking[cardanoNetwork]);
+    const trezorDRepBech32 = trezorDRep?.drep.bech32;
+
+    return (
+        <NewModal onCancel={onCancel}>
+            <StyledH2>
+                <FormattedMessage
+                    id="TR_CARDANO_WITHDRAW_MODAL_TITLE"
+                    defaultMessage="Delegate voting rights"
+                />
+            </StyledH2>
+            <Row>
+                <FormattedMessage
+                    id="TR_CARDANO_WITHDRAW_MODAL_DESCRIPTION"
+                    defaultMessage="When withdrawing your rewards, you can choose to support the Cardano ecosystem by delegating your community voting rights. Your votes will be delegated to a representative of Five Binaries, the operators of Cardano staking in Trezor Suite. This helps strengthen the network's resilience, sustainability, and community-driven governance. If you prefer, you can easily opt out of governance."
+                />
+            </Row>
+            <Row>
+                <Content>
+                    <Column>
+                        <Title>
+                            <FormattedMessage
+                                id="TR_CARDANO_WITHDRAW_MODAL_SUB_TITLE"
+                                defaultMessage="Delegate Representative (DRep)"
+                            />
+                        </Title>
+                        <HiddenPlaceholder>
+                            <Value>
+                                {trezorDRepBech32}
+                                <Link href={`https://gov.tools/drep_directory/${trezorDRepBech32}`}>
```

Comment:
I am slightly unhappy about the fact that trezorDRepBech32 can be undefined per typescript. Do you think it would make sense to add an early return above to handle this case?